File: app/src/main/java/com/ethan/ceiling/adapter/AdapterMallViewPage2.java
package com.ethan.ceiling.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.ethan.ceiling.ui.fragment.FragmentMallShoppingList;

import java.util.List;

public class AdapterMallViewPage2 extends FragmentStateAdapter {

    private List<String> list;
    public AdapterMallViewPage2(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle, List<String> list) {
        super(fragmentManager, lifecycle);
        this.list =list;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return FragmentMallShoppingList.newInstance(list.get(position),"test");
    }

    @Override
    public int getItemCount() {
        return list.size();
    }
}

File: app/src/main/java/com/ethan/ceiling/ui/adapter/AdapterMallViewPage2.kt
package com.ethan.ceiling.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.ethan.ceiling.ui.fragment.FragmentMallShoppingList.Companion.newInstance

class AdapterMallViewPage2(fragmentManager: FragmentManager, lifecycle: Lifecycle, private val list: List<String>) : FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment {
        return newInstance(list[position], "test")
    }

    override fun getItemCount(): Int {
        return list.size
    }
}
